### PR TITLE
ddl: rewrite some cancel tests to an unified test

### LINKED ISSUE #32939: Refine cancel in test
## Enhancement
We use `admin.CancelJobs(txn, test.jobIDs)` in test to cancel a DDL job.
The code is a little complex and doesn't compatible with the new DDL framework.
We should use SQL statement "admin cancel ddl job" to achieve the same goal.

### PR DESCRIPTION
**Pull request by wjhuang2016**
Signed-off-by: wjhuang2016 <huangwenjun1997@gmail.com>

### What problem does this PR solve?

Issue Number: close #32939

Problem Summary:
Rewrite some cancel tests to an unified test.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: ddl/cancel_test.go [hunk 1]
```diff
@@ -195,10 +231,11 @@ func TestCancel(t *testing.T) {
 	hook := &ddl.TestDDLCallback{Do: dom}
 	i := 0
 	cancel := false
+	cancelWhenReorgNotStart := false
 
 	hookFunc := func(job *model.Job) {
 		if job.SchemaState == allTestCase[i].cancelState && !cancel {
-			if job.SchemaState == model.StateWriteReorganization && ddl.MayNeedReorg(job) && job.RowCount == 0 {
+			if !cancelWhenReorgNotStart && job.SchemaState == model.StateWriteReorganization && ddl.MayNeedReorg(job) && job.RowCount == 0 {
```
Comment (tangenta) [inline ddl/cancel_test.go:238]: `cancelWhenReorgNotStart` is always `true` after the first test case is run. Please check if this is expected.

Comment (wjhuang2016) [reply ddl/cancel_test.go:239]: Good catch, fixed.

File: ddl/cancel_test.go [hunk 2]
```diff
@@ -73,16 +73,33 @@ var allTestCase = []testCancelJob{
...
 	{"alter table t drop column c3", false, model.StateDeleteReorganization, true, true, []string{"alter table t add column c3 bigint"}},
 	{"alter table t drop column c3", false, model.StatePublic, false, true, []string{"alter table t add column c3 bigint"}},
+	// Drop column with index.
+	{"alter table t drop column c3", true, model.StateNone, true, false, []string{"alter table t add column c3 bigint", "alter table t add index idx_c3(c3)"}},
+	{"alter table t drop column c3", false, model.StateDeleteOnly, true, false, nil},
+	{"alter table t drop column c3", false, model.StateDeleteOnly, false, true, []string{"alter table t add column c3 bigint", "alter table t add index idx_c3(c3)"}},
+	{"alter table t drop column c3", false, model.StateWriteOnly, true, true, []string{"alter table t add column c3 bigint", "alter table t add index idx_c3(c3)"}},
+	{"alter table t drop column c3", false, model.StateDeleteReorganization, true, true, []string{"alter table t add column c3 bigint", "alter table t add index idx_c3(c3)"}},
+	{"alter table t drop column c3", false, model.StatePublic, false, true, []string{"alter table t add column c3 bigint", "alter table t add index idx_c3(c3)"}},
```
Comment (tangenta) [inline ddl/cancel_test.go:102]: Why can't this DDL be canceled? The state is public.

I have the same question for line 95.

**Review by tangenta — APPROVED** [review]

File: ddl/cancel_test.go (hunk 2 above)
Comment (wjhuang2016) [reply ddl/cancel_test.go:107]: Added TODO to fix them.

**Review by xiongjiwei — APPROVED** [review]

**Pull request merged**